v1/hash.go

```diff
@@ -27,18 +27,18 @@ import (
 
 // Hash is an unqualified digest of some content, e.g. sha256:deadbeef
 type Hash struct {
-	algorithm string
-	hex       string
+	A string
+	H string
```

Comment:
Alg, Hex? I know that lots of golang people prefer... _condensed_ labels, but you're triggering my front-end development PTSD

Is this meant to encourage use of the methods while preventing cmp.Diff from complaining about unexported fields? iirc, there's an argument you can pass to it to prevent that as well.

Reply:
My preference would be `Algorithm` and `Hex`, but those are taken by methods.

Yes, this is for `cmp.Diff`.  This seemed the lesser evil than passing a "don't panic" flag everywhere.

Reply:
Do they need to be methods?

Reply:
Done